File: Search a 2D Matrix.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int> > &matrix, int target) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        if(matrix.empty())
            return false;
        
        int colum = matrix[0].size();
        int row = matrix.size();
        int i = 0;
        int j = colum - 1;
        while(i < row && j > -1)
        {
            if(matrix[i][j]==target)
            {
                return true;
            }
            else if(matrix[i][j] > target)
            {
                j--;
            }
            else
            {
                i++;
            }
        }
        
        return false;
    }
};

File: Search a 2D Matrix.java
public class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        // Start typing your Java solution below
        // DO NOT write main() function
        if(matrix.length == 0||matrix[0].length == 0)
        {
            return false;
        }
        boolean ret = false;
        int i = 0;
        int j = matrix[0].length - 1;
        while(i < matrix.length&& j > -1)
        {
            if(matrix[i][j]==target)
            {
                ret = true;
                break;
            }
            else if(matrix[i][j] > target)
            {
                j--;
            }
            else
            {
                i++;
            }
        }    
        
        return ret;
    }
    
}